forte/data/data_store.py

```diff
@@ -315,12 +327,16 @@ def set_attribute(self, tid: int, attr_name: str, attr_value: Any):
             attr_name (str): Name of the attribute.
             attr_value (any): Value of the attribute.
         """
-        if tid not in self.__entry_dict:
-            raise KeyError(f"Entry with tid {tid} not found.")
-        entry_type = self.__entry_dict[tid][self.__entry_type_idx]
-        if attr_name not in self._type_attributes[entry_type]:
-            raise ValueError(f"{entry_type} has no {attr_name} attribute.")
-        attr_id = self._type_attributes[entry_type][attr_name]
+        try:
+            entry_type = self.__entry_dict[tid][constants.ENTRY_TYPE_INDEX]
+        except KeyError as e:
+            raise KeyError(f"Entry with tid {tid} not found.") from e
```

Comment:
Better if we add `Exception` into the docstring.

Reply:
https://github.com/asyml/forte/blob/master/forte/pipeline.py#L498, this is how we write exceptions